### PR TITLE
Feature/recommended appointment scheduling

### PR DESCRIPTION
**Pull request by Stefann11**

### CODE REVIEW

File: PatientWebApplication/HealthClinic.DAL/Service/DoctorService.cs [hunk 1]
```diff
@@ -205,7 +205,10 @@ public Boolean DoesDoctorHaveAnAppointmentAtSpecificTime(DoctorUser doctor, Time
 
             foreach (DoctorAppointment appointment in listOfAppointments)
             {
-                if (areDatesEqual(appointment.Date, date) && checkIfDoctorIsBusyForAppointment(appointment, time)) return true;
+                if (!appointment.IsCanceled)
+                {
+                    if (areDatesEqual(appointment.Date, date) && checkIfDoctorIsBusyForAppointment(appointment, time)) return true;
+                }
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/DoctorService.cs:211]: Merge if statements into one.

**Review by trki293 — CHANGES_REQUESTED** [review]
Code changes required.

File: PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs [hunk 2]
```diff
@@ -123,15 +133,81 @@ public List<DoctorAppointment> GetAppointmentsForDoctor(int id)
...
+                    {
+                        availableAppointments.Add((new DoctorAppointment(0, time, MakeStringFromDate(date), _patientRepository.Find(patientId), doctorService.GetByid(doctorId), new List<Referral>(), doctorService.GetByid(doctorId).ordination)));
+                        if (availableAppointments.Count >= 5)
+                        {
+                            return availableAppointments;
+                        }
+                    }
+                    time = time.Add(TimeSpan.FromMinutes(15));
+                }
+            }
```
Comment (trki293) [inline PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs:206]: It is necessary to refactor the function. Separate into several smaller functions and call to the main.

**Review by trki293 — APPROVED** [review]
PR is approved.

File: PatientWebApplication/PatientWebApplicationTests/GetRecommendedAppointmentTests.cs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            stubRepository.Setup(m => m.GetByid(2)).Returns(doctor2);
+            stubRepository.Setup(m => m.GetByid(3)).Returns(doctor3);
+
+            return stubRepository.Object;
+        }
+
+        private static IPatientsRepository CreatePatientStubRepository()
+        {
+            var stubRepository = new Mock<IPatientsRepository>();
+            List<PatientUser> patients = new List<PatientUser>();
```
Comment (velicko-tam) [inline PatientWebApplication/PatientWebApplicationTests/GetRecommendedAppointmentTests.cs:199]: Remove unnecessary List<PatientUser> patients

File: PatientWebApplication/PatientWebApplicationTests/ScheduleRecommendedAppointmentTests.cs [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+            stubRepository.Setup(m => m.GetByid(2)).Returns(doctor2);
+            stubRepository.Setup(m => m.GetByid(3)).Returns(doctor3);
+
+            return stubRepository.Object;
+        }
+
+        private static IPatientsRepository CreatePatientStubRepository()
+        {
+            var stubRepository = new Mock<IPatientsRepository>();
+            List<PatientUser> patients = new List<PatientUser>();
```
Comment (velicko-tam) [inline PatientWebApplication/PatientWebApplicationTests/ScheduleRecommendedAppointmentTests.cs:151]: Remove unnecessary patients list

**Review by velicko-tam — CHANGES_REQUESTED** [review]
PR is not approved. Marked parts need to be fixed.

**Review by velicko-tam — APPROVED** [review]
PR is approved. Clean code principles were implemented.

**Pull request merged**